frontend/packages/shared/src/hooks/queries/useOptionListsQuery.ts

```diff
@@ -2,19 +2,33 @@ import { useServicesContext } from 'app-shared/contexts/ServicesContext';
 import { QueryKey } from 'app-shared/types/QueryKey';
 import type { UseQueryResult } from '@tanstack/react-query';
 import { useQuery } from '@tanstack/react-query';
+import type { Option } from 'app-shared/types/Option';
+import type { OptionsLists } from 'app-shared/types/api/OptionsLists';
 
-export const useOptionListsQuery = (org: string, app: string): UseQueryResult<string[]> => {
+export const useOptionListsQuery = (
+  org: string,
+  app: string,
+): UseQueryResult<Map<string, Option[]>> => {
   const { getOptionLists } = useServicesContext();
-
-  return useQuery<any>({
+  return useQuery<Map<string, Option[]>>({
     queryKey: [QueryKey.OptionLists, org, app],
     queryFn: () =>
       getOptionLists(org, app).then((result) => {
-        const optionLists = {};
-        Object.keys(result).forEach((optionListId) => {
-          optionLists[optionListId] = result[optionListId];
-        });
-        return optionLists;
+        return convertToMap(result);
       }),
   });
 };
+
+function convertToMap(result: OptionsLists): Map<string, Option[]> {
+  const optionsMap = new Map<string, Option[]>();
```

Comment:
Do we need to convert it to Map? I thought it was okay to use record all the way and access the correct optionList in the editor by iterating? 🙈